lib/active_merchant/billing/gateways/mondido.rb

```diff
@@ -0,0 +1,658 @@
+require 'openssl'
+require 'digest'
+require 'base64'
+require 'json'
+
+module ActiveMerchant #:nodoc:
+  module Billing #:nodoc:
+    class MondidoGateway < Gateway
+      self.display_name = 'Mondido'
+      self.homepage_url = 'https://www.mondido.com/'
+      self.live_url = 'https://api.mondido.com/v1/'
+
+      # For the entire accepted currencies list from Mondido, please access:
+      # http://doc.mondido.com/api#currencies
+      self.default_currency = 'USD'
+
+      # :dollars  => decimal (e.g., 10.00)
+      # :cents    => integer with cents (e.g., 1000 = $10.00)
+      self.money_format = :dollars
+
+      self.supported_countries = %w(AT BE BG HR CY CZ DK EE FI FR DE GI GR HU IS IM IT LV LI 
+        LT LU MC NL NO PL PT IE MT RO SK SI ES SE CH GB)
+
+      self.supported_cardtypes = [:visa, :master, :discover, :american_express, 
+          :diners_club, :jcb, :switch, :solo, :maestro, :laser]
+
+      # Mapping of CVV check result codes from Mondido to CVVResult standard codes
+      # For more codes, please check the CVVResult class
+      CVC_CODE_TRANSLATOR = {
+        'errors.card_cvv.missing' => 'S', # CVV should have been present
+        'errors.card_cvv.invalid' => 'N', # CVV does not match
+      }
+
+      # Mapping of error codes from Mondido to Gateway class standard error codes
+      STANDARD_ERROR_CODE_TRANSLATOR = {
+        'errors.card_number.missing' => STANDARD_ERROR_CODE[:invalid_number],
+        'errors.card_number.invalid' => STANDARD_ERROR_CODE[:invalid_number],
+        'errors.card_expiry.missing' => STANDARD_ERROR_CODE[:invalid_expiry_date],
+        'errors.card_expiry.invalid' => STANDARD_ERROR_CODE[:invalid_expiry_date],
+        'errors.card_cvv.missing' => STANDARD_ERROR_CODE[:invalid_cvc],
+        'errors.card_cvv.invalid' => STANDARD_ERROR_CODE[:invalid_cvc],
+        'errors.card.expired' => STANDARD_ERROR_CODE[:expired_card],
+        'errors.zip.missing' => STANDARD_ERROR_CODE[:incorrect_zip],
+        'errors.address.missing' => STANDARD_ERROR_CODE[:incorrect_address],
+        'errors.city.missing' => STANDARD_ERROR_CODE[:incorrect_address],
+        'errors.country_code.missing' => STANDARD_ERROR_CODE[:incorrect_address],
+        'errors.payment.declined' => STANDARD_ERROR_CODE[:card_declined],
+        'errors.unexpected' => STANDARD_ERROR_CODE[:processing_error]
+      }
+
+      def initialize(options={})
+        requires!(options, :merchant_id, :api_token, :hash_secret)
+
+        @merchant_id = options[:merchant_id]
+        @api_token = options[:api_token]
+        @hash_secret = options[:hash_secret]
+
+        # Optional: RSA Encryption
```

Comment:
Take Heartbleed vulnerability as an example. You'll be able to read the remote server's memory. What happens if the value that you read is encrypted? It's just defense in depth, but yes, you're right about the complexity. Plus the server may have its security requirements reduced as it will recept encrypted data. Depends on more factors, but that's good for my explanation I think.